src/osd/OSDHealthMetric.h

```diff
@@ -0,0 +1,94 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <tuple>
+#include "include/health.h"
+#include "include/denc.h"
+
+enum class osd_metric : uint16_t {
+  NUM_SLOW_OPS,
+  OLDEST_BLOCKED_OP,
+  NUM_PENDING_CREATING_PGS,
+  NONE,
+};
+
+using osd_metric_accumulate_t = std::function<uint32_t(uint32_t, uint32_t)>;
+
+inline std::tuple<const char* /* code */,
+	   health_status_t /* sev */ ,
+	   const char* /* summary */,
+	   osd_metric_accumulate_t,
+	   uint32_t /* init */>
```

Comment:
I think we need to have multiple values and multiple accumulators within a single health check, so that we can have a single "slow requests" check that has both the total number and also the oldest time